suse_migration_services/migration_config.py

```diff
@@ -106,3 +117,11 @@ def is_zypper_migration_plugin_requested(self):
     def _write_config_file(self):
         with open(self.migration_config_file, 'w') as config:
             yaml.dump(self.config_data, config, default_flow_style=False)
+
+    def _read_baseproduct(self):
+        xml = ElementTree()
+        root_path = Defaults.get_system_root_path()
+        baseproduct_metadata = os.sep.join(
+            [root_path, 'etc', 'products.d', 'baseproduct']
+        )
+        return xml.parse(baseproduct_metadata)
```

Comment:
This is not safe, there is no guarantee that there is a baseproduct link. Again see units/product_setup.py where we already solved this problem... actually you wrote that code a while back ;)